File: C++ Program/OOP.cpp
//Class = characteristics + behaviour
//Class = Data + Function

#include<iostream>
using namespace std;

class Demo 
{
    public:
        int No1;
        int No2;

        void fun()
        {
            cout<<"Inside fun\n";
        }
};

int main()
{
    Demo obj1 ;
    Demo obj2 ;

    cout<<"Size of object : "<<sizeof(obj1)<<"\n";   //8
    obj1.No1 =10;
    obj1.No2 =20;

    obj2.No1 =30;
    obj2.No2 =40;

    obj1.fun();
    obj2.fun();

    cout<<obj1.No2<<"\n";    //20

    return 0;
}


/*

C:\Users\admin\Desktop\PPA\C++Programming>g++ OOP.cpp -o Myexe

C:\Users\admin\Desktop\PPA\C++Programming>Myexe
Size of object : 8
Inside fun
Inside fun
20


*/

File: Java Program/OOP.java


class OOP 
{
    public static void main(String Arg[])
    {
        System.out.println("Inside main function");

        Arithmatic obj1 = new Arithmatic ();
        Arithmatic obj2 = new Arithmatic (11,10);

        int iRet = 0;

        iRet = obj1.Addition();       //0
        System.out.println("Addition is : "+iRet);

        iRet = obj1.Substraction();    //0
        System.out.println("Substraction is : "+iRet);

        iRet = obj2.Addition();        //21
        System.out.println("Addition is : "+iRet);

        iRet = obj2.Substraction();      //1
        System.out.println("Substraction is : "+iRet);

    }
}


class Arithmatic
{
    public int No1;
    public int No2;

    public Arithmatic()       //Default constructor
    {
        System.out.println("Inside default constructor");
        this.No1 = 0 ;
        this.No2 = 0 ;
    }
    public Arithmatic(int i , int j)     //Parameterised constructor
    {
        System.out.println("Inside parameterised constructor");
        this.No1 = i ;
        this.No2 = j ;
    }

    public int Addition()
    {
        int Ans = 0;
        Ans = this.No1 + this.No2;
        return Ans;
    }

    public int Substraction()
    {
        int Ans = 0;
        Ans = this.No1 - this.No2;
        return Ans;
    }
}



/*

C:\Users\admin\Desktop\PPA\Java Programming>javac OOP.java

C:\Users\admin\Desktop\PPA\Java Programming>java OOP.java
Inside main function
Inside default constructor
Inside parameterised constructor
Addition is : 0
Substraction is : 0
Addition is : 21
Substraction is : 1

*/